onboarding/mlsys/1-backprop.md

```diff
@@ -0,0 +1,414 @@
+# Part 1: Backpropagation
+
+Machine learning is all about getting computers to solve math problems for you.
+
+Specifically, it's about solving math problems like, "For a given mathematical
+model, what weights in the model would cause the model to calculate the most
+accurate predictions for a set of inputs?"
+
+You're going to write code to solve this exact problem for one particularly
+simple model.
+
+Afterwards, you will understand of basic concepts like gradient descent and
+batch size.  You will also understand how those basic concepts translate into
+some of Determined's advanced features, like gradient aggregation,
+data-parallel distributed training, and model-parallel distributed training.
+
+## The Model
+
+Our model is going to be be a simple linear model: `ypred = m*x` (`pred` is
+short for "predicted"). In statistics they would say there is no bias term, or
+in algebra they would there's no constant term.  Just a stright line through
+the origin.
+
+## The Data Set
+
+Our made-up dataset has 4 data points:
+
+- `(x=1, ylabel=1)`
```

Comment:
Nit: might not want to call these labels everywhere and just have them be arbitrary output values.

The word "label" implies categorical which implies a different typical loss function and problem setup. Not a big deal, but might be disorienting to anyone who knows a bit more ML or confusing when people try to map the problem setup here to the setup in classification problems they might see in the code base or elsewhere.

Reply:
I'll go with `ytrue`/`ypred` to avoid the categorical implication.  I like avoid the unmarked `y` because I want to make sure every instance of `y` reminds the reader that there's two kinds of `y` here.